pytest_html/plugin.py

```diff
@@ -127,18 +134,41 @@ def __lt__(self, other):
                      'XPassed', 'Skipped', 'Passed')
             return order.index(self.outcome) < order.index(other.outcome)
 
-        def append_extra_html(self, extra, additional_html, links_html):
+        def append_extra_html(self, extra, extra_index, test_index):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                href = '#'
-                image = 'data:image/png;base64,{0}'.format(
-                        extra.get('content'))
-                additional_html.append(html.div(
-                    html.a(html.img(src=image), href="#"),
+                if self.self_contained:
+                    image_src = 'data:image/png;base64,{0}'.format(
+                            extra.get('content'))
+                    href = '#'
+                else:
+                    hash_key = "{0}{1}{2}".format(
+                               self.test_id, str(extra_index),
+                               str(test_index)).encode('utf-8')
```

Comment:
If we're just concatenating these, try using `''.join([self.test_id, extra_index, test_index])`